controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -844,6 +889,92 @@ bool wfa_ca::get_send_1905_1_tlv_hex_list(std::list<tlv_hex_t> &tlv_hex_list,
     return false;
 }
 
+/**
+ * @brief Writes hex string into the class buffer.
+ * 
+ * @param[in] value string in hex notation.
+ * @param[out] length Length of current tlv.
+ * @return false if length smaller than data, else true.
+ */
+bool tlvPrefilledData::add_tlv_value_hex_string(const std::string &value, uint16_t &length)
+{
+    // iterate every two chars (1 octet) and write it to buffer
+    // start with char_idx = 2 to skip "0x"
+    for (size_t char_idx = 2; char_idx < value.size(); char_idx += 2) {
+        if (length == 0) {
+            return false;
+        }
+        *m_buff_ptr__ = std::stoi(value.substr(char_idx, 2), nullptr, 16);
+        m_buff_ptr__++;
+        length--;
+    }
+    return true;
+}
+
+/**
+ * @brief Writes a decimal string into the class buffer.
+ * 
+ * @param[in] value Decimal string.
+ * @param[out] length Length of current tlv.
+ * @return false if length smaller than data, else true.
+ */
+bool tlvPrefilledData::add_tlv_value_decimal_string(const std::string &value, uint16_t &length)
+{
+    int i_dec        = std::stoi(value.substr(2, value.length()));
+    int num_of_bytes = i_dec / UINT8_MAX + (i_dec % UINT8_MAX != 0 ? 1 : 0);
+    std::string s    = string_utils::int_to_hex_string(i_dec, num_of_bytes);
```

Comment:
The idea was to avoid converting back to string...

If you do convert back to string, at least you should reuse the `add_tlv_value_hex_string` function (so prepend `0x`).

But I'd rather to see it done explicitly here (which is a bit more difficult, I admit, because it's big-endian).

```suggestion
    for (size_t idx = 0; idx < num_of_bytes; idx++) {
        m_buff_ptr__[num_of_bytes - 1 - idx] = i_dec % 256;
        i_dec /= 256;
    }
    m_buff_ptr__ += num_of_bytes;
    length -= num_of_bytes;
```

Or, if you want to avoid the inductive approach

```suggestion
    for (size_t idx = 0; idx < num_of_bytes; idx++) {
        *m_buff_ptr__ = i_dec / (1<<(8*idx));
        m_buff_ptr__++;
        length--;
    }
```